### ISSUE TITLE
Wrapper logic: fre pp checkout

### ISSUE THREAD

**Issue by cwhitlock-NOAA** [NOAA-GFDL/fre-cli#174]
Preamble: 
The thing-currently-named-wrapper (wrapper.py) works for testing fre-cli, when we have all history files already available at the time of running. However, when we get to running production jobs, the models will be sending over bundles of history files and post-processing them in parallel. This breaks some of the logic currently in the wrapper flow - in particular, the assumptions that there's not already a pre-existing experiment belonging to the same user that the current set of history files is being added to, and the assumption that there's not already an experiment with that name running. 

The logic we need is encapsulated in a flowchart at the end of this issue; this breaks it down by tool.

The tool: 
Fre pp checkout currently exits with an error when it detects a prior experiment with the same name belonging to the same user. It needs to be able to see if the prior experiment uses the same checkout specs (branch + commit history) 

fre pp checkout:

[ ] Can we check out a branch other than main?
   - [ ] test: can we check out a non-main branch
[ ] Is there a preexisting experiment with the same -e -p -t as the current experiment?
   - [ ] test: can we detect a preexisting experiment
[ ] If so, do the remote and the local / branch specifications match?   
   - [ ] test: can we detect the branch specification
[ ] If so, has the repo been updated since the prior experiment was run? Fail with a good error in that case, or tell your users REALLY LOUDLY that the results may not match your prior experiment chunk.
   - [ ] test: can it tell that your repo and the remote repo are out of sync? I think that we need a persistent protected branch for this testing to work, and possibly a prior experiment owned by the github user that is permanently out of sync with a prior experiment.

In practice, I think the tests for items 2 and 3 can be combined together into a single test that checks for a preexisting experiment and does the thing if so. (note to self: any logic that starts with "if so" could probably get a single test)

![image](https://github.com/user-attachments/assets/e79fd580-0a80-4814-a431-9f0f9956fab2)

**Comment by kiihne-noaa** [comment]
have made a branch for these changes, and currently pushed a change that adds an exit error if the local branch is out of date, and or different from the remote branch. Also added in 'forced' option that overrules this branch check. Can be found at https://github.com/NOAA-GFDL/fre-cli/commit/e063d85868d439378c84df1fca729364dd884f62

**Comment by kiihne-noaa** [comment]
Thank you for catching that, it was a syntax error as I changed that variable name several times. it is now corrected to be branch_status, which is defined in line 35 as the output from a `git status` command.

**Comment by ilaflott** [comment]
@cwhitlock-NOAA this issue appears to supersede #44 , can i close #44 safely?

**Comment by ilaflott** [comment]
apologies for accidentally closing THIS issue -_-...

**Comment by singhd789** [comment]
Some of the bullets in the description might be done already but there could be some good points to keep in mind with `fre workflow checkout` development. I'll add this issue to the description of this issue, #697 for things to keep in mind. 

In general, I think this issue description/checkout tool functionality will be included in issue #697 , so I will close this issue as being superseded by #697 697
